docs/sql-ref-functions-udf-aggregate.md

```diff
@@ -1,22 +1,133 @@
 ---
 layout: global
-title: User defined Aggregate Functions (UDAF)
-displayTitle: User defined Aggregate Functions (UDAF)
+title: User Defined Aggregate Functions (UDAFs)
+displayTitle: User Defined Aggregate Functions (UDAFs)
 license: |
   Licensed to the Apache Software Foundation (ASF) under one or more
   contributor license agreements.  See the NOTICE file distributed with
   this work for additional information regarding copyright ownership.
   The ASF licenses this file to You under the Apache License, Version 2.0
   (the "License"); you may not use this file except in compliance with
   the License.  You may obtain a copy of the License at
- 
+
      http://www.apache.org/licenses/LICENSE-2.0
- 
+
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+
+User-Defined Aggregate Functions (UDAFs) are user-programmable routines that act on multiple rows at once and return a single aggregated value as a result. This documentation lists the classes that are required for creating and registering UDAFs. It also contains examples that demonstrate how to define and register UDAFs in Scala and invoke them in Spark SQL.
+
+### Aggregator[-IN, BUF, OUT]
+
+A base class for user-defined aggregations, which can be used in Dataset operations to take all of the elements of a group and reduce them to a single value.
+- IN The input type for the aggregation.
```

Comment:
nit: need a single blank?
```
A base class for user-defined aggregations, which can be used in Dataset operations to take all of the elements of a group and reduce them to a single value.

- IN The input type for the aggregation.
```

Reply:
nit: also, how about putting a separator between `IN` and the statement? e.g., `IN - The input...`

Reply:
btw, the contents are somewhat overlapped with https://people.apache.org/~pwendell/spark-nightly/spark-master-docs/latest/sql-programming-guide.html#type-safe-user-defined-aggregate-functions ?

Reply:
btw, the contents are somewhat overlapped with https://spark.apache.org/docs/3.0.0-preview/sql-getting-started.html#aggregations ? Probably, we need to organize these aggregator contents.

Reply:
I will think about this.

Reply:
I personally think its better to move all the contents of `sql-getting-started.html` into the SQL doc. Since the user-defined aggregator topic is advanced one, the SQL doc is more suitable than the `Getting Started` doc. cc: @gatorsmile

Reply:
@maropu sounds good to me.